Sources/armory/logicnode/GetCursorStateNode.hx

```diff
@@ -13,7 +13,7 @@ class GetCursorStateNode extends LogicNode {
 
 		return switch (from) {
 			case 0:
-				mouse.hidden ? return false : mouse.locked ? return false : return true;
+				mouse.hidden ? mouse.locked ? return true : return false;
```

Comment:
What exactly is happening here? It looks like you return something inside of another return statement (the one before the `switch`). Also, the two nested ternary if statements might be a bit confusing, maybe you could use parentheses to make it clear how the nesting works.

Reply:
There is some way to test all .hx files at once without add them in the node tree? I would like to make sure that I haven't touched anything I shouldn't have.

Reply:
It looks like it just returns whether the mouse is hidden and locked, right? Then you can just write `mouse.hidden && mouse.locked`. The `return` statement comes before the switch, so there is no need to use it again.

What I mean with nesting and parentheses is the following:

```haxe
cond1 ? (cond2 ? doSomething2() : doSomethingElse2()) : doSomethingElse1();
```

This syntax makes it clear which action belongs to which condition. If there are no brackets/parantheses, it can be a bit difficult to see what is going on.

> Is there some way to test all .hx files at once without add them to the node tree? I would like to make sure that I haven't touched anything I shouldn't have.

Yes. This is called unit testing, but Armory doesn't really do that (see https://github.com/armory3d/armory/issues/864). Unit tests usually test certain functions by giving them example inputs and test whether the output is correct ([simple example in Haxe](https://gitlab.com/koui/Koui/-/blob/master/Tests/elements/TestTextInput.hx)). This can be automated to be run on each merge request for example.